browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -7002,5 +7004,17 @@ L.MessageStore = L.Class.extend({
 		Object.keys(othersMessages).forEach(function (msgType) {
 			othersMessages[msgType].forEach(callback);
 		});
+	},
+
+	_cleanUpSelectionMessages: function(messages) {
+		// must be called only from _replayPrintTwipsMsg !!
+		// check if textselection is empty
+		// if it is, we need to handle textselectionstart and textselectionend
+		// otherwise we get handles without selection and they also may appear in the wrong cell
+		// but it is also reproducible on the same cell too. e.g. selection handles without selection
+		if (!messages && !messages['textselection'] && messages['textselection'] !== 'textselection: ')
+			return;
+		messages['textselectionstart'] = 'textselectionstart: ';
+		messages['textselectionend'] = 'textselectionend: ';
```

Comment:
I'm wondering if we cannot do something in code which handles these messages?
(so we will keep message store clean from message modifications)

Reply:
the messages is bind to _onMessage through forEach method and it is handled there only. Unfortunately we cannot check the state there because same function is used when we actually get the message from the API as well and we may not be sure about it.